internal/cephfs/nodeserver.go

```diff
@@ -694,6 +728,34 @@ func (ns *NodeServer) NodeGetVolumeStats(
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	// health check first, return without stats if unhealthy
+	healthy, msg := ns.healthChecker.IsHealthy(req.GetVolumeId(), targetPath)
+
+	// If healthy and an error is returned, it means that the checker was not
+	// started. This could happen when the node-plugin was restarted and the
+	// volume is already staged and published.
+	if healthy && msg != nil {
+		// Start a StatChecker for the mounted targetPath, yhis prevents
+		// writing a file in the user-visible location. Ideally a (shared)
+		// FileChecker is started with the stagingTargetPath, but we can't
+		// get the stagingPath from the request.
+		err = ns.healthChecker.StartChecker(req.GetVolumeId(), targetPath, hc.StatCheckerType)
```

Comment:
i also see a very small race for the below case

* During NodeUnpublish cephcsi stopped the health checker
* Before umount operation we got a `NodeGetVolumeStats` RPC call which inturn started a health checker
* Now we have a stale health check which will be cleanedup only during NodeUnstage (i think)

Is it possible to have such a case?

Reply:
I do not expect the CO to request `NodeUnpublish` and then still use the same `publishTargetPath` in a later `NodeGetVolumeStats`. If a CO does that, then yes, there would be a stale health checker (if it was started in a `NodeGetVolumeStats` call, and not in `NodeStageVolume`).